src/main/java/com/example/Restaurant_Management_System_REST_API/model/entity/Customer.java

```diff
@@ -0,0 +1,24 @@
+package com.example.Restaurant_Management_System_REST_API.model.entity;
+
+import com.example.Restaurant_Management_System_REST_API.model.ContactDetails;
+import jakarta.persistence.*;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.time.LocalDateTime;
+
+@Entity
+@Getter
+@Setter
+public class Customer {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id", nullable = false)
+    private Long id;
+    private LocalDateTime creationTime;
+    @OneToOne(mappedBy = "customer")
+    private Reservation reservation;
```

Comment:
Could you describe for me this 21st line of code? What purpose and what goal it achieves?

Reply:
There will be a role CUSTOMER so in the database personnel can check if a specific customer has any reservation

Reply:
ps. I know it is not in diagram, I tried to update it but I cannot reopen it on draw.io -> I got the error and I don't want to do this diagram again!!!!

Reply:
Okay, I'll leave it as it is, but I can see an issue here in future development, and I didn't mean any diagram.
If this occur, I'll come back to this conversation. Feel free to resolve this convo.